src/strtab.rs

```diff
@@ -60,37 +55,54 @@ impl<'a> Strtab<'a> {
                 overflow
             )));
         }
-        Ok(Strtab {
-            bytes: &bytes[offset..end],
-            delim: ctx::StrCtx::Delimiter(delim),
-        })
+        let mut result = Self::from_slice_unsafe(bytes, offset, len, delim);
+        {
+            let mut i = 0;
+            while i < len {
+                let string = get_str(i, result.bytes, result.delim)?;
+                result.strings.push((i, string));
+                i += string.len() + 1;
+            }
+        }
+        Ok(result)
     }
     #[cfg(feature = "alloc")]
-    /// Converts the string table to a vector, with the original `delim` used to separate the strings
-    pub fn to_vec(&self) -> error::Result<Vec<&'a str>> {
-        let len = self.bytes.len();
-        let mut strings = Vec::with_capacity(len);
-        let mut i = 0;
-        while i < len {
-            let string = self.get(i).unwrap()?;
-            i = i + string.len() + 1;
-            strings.push(string);
-        }
-        Ok(strings)
+    /// Parses a `Strtab` with `bytes` as the backing string table, using `delim` as the delimiter between entries.
+    pub fn new(bytes: &'a [u8], delim: u8) -> error::Result<Self> {
+        Self::from_slice(bytes, 0, bytes.len(), delim)
     }
-    /// Safely parses and gets a str reference from the backing bytes starting at byte `offset`.
+    #[cfg(feature = "alloc")]
+    /// Converts the string table to a vector of parsed strings.
+    pub fn to_vec(&self) -> Vec<&'a str> {
+        self.strings.iter().map(|&(_key, value)| value).collect()
+    }
+    /// Safely gets a str reference from the parsed table starting at byte `offset`.
+    ///
     /// If the index is out of bounds, `None` is returned.
     /// Requires `feature = "alloc"`
     #[cfg(feature = "alloc")]
```

Comment:
it's curious this method (`get`) required `alloc` feature before, but i suppose it's a happy accident since now you really are allocating if you use this `get`

Reply:
Probably because `map_err(core::convert::Into::into)` allocated an error message `String`.